src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/RestCallHelper.java

```diff
@@ -23,78 +27,120 @@
+                .header(HEADER_PAGE_LIMIT, String.valueOf(limit))
+                .attributes(clientRegistrationId("elite2-api"))
+                .retrieve()
+                .bodyToMono(responseType)
+                .block();
     }
 
-    protected <T> ResponseEntity<T> getWithPagingAndSorting(final URI uri, final PagingAndSortingDto pagingAndSorting,
-                                                            final ParameterizedTypeReference<T> responseType, final boolean admin) {
-        return getRestTemplate(admin).exchange(
-                uri.toString(),
-                HttpMethod.GET,
-                withPagingAndSorting(pagingAndSorting),
-                responseType);
+    <T> ResponseEntity<T> getEntityWithPagingAndSorting(final String path,
+                                                        final MultiValueMap<String, String> queryParams,
+                                                        final Map<String, String> uriVariables,
+                                                        final PagingAndSortingDto pagingAndSorting,
+                                                        final ParameterizedTypeReference<T> responseType,
+                                                        final boolean admin) {
+        return getWebClient(admin)
+                .get()
+                .uri(uriBuilder -> uriBuilder.path(path).queryParams(queryParams).build(uriVariables))
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                .headers(withPagingAndSorting(pagingAndSorting))
+                .attributes(clientRegistrationId("elite2-api"))
+                .retrieve()
+                .toEntity(responseType)
+                .block();
     }
 
-    protected <T> ResponseEntity<T> getWithPaging(final URI uri, final PagingAndSortingDto pagingAndSorting,
-                                                  final ParameterizedTypeReference<T> responseType) {
-        return restTemplate.exchange(
-                uri.toString(),
-                HttpMethod.GET,
-                withPaging(pagingAndSorting),
-                responseType);
+    <T> ResponseEntity<T> getEntityWithPaging(final String path,
+                                              final MultiValueMap<String, String> queryParams,
+                                              final Map<String, String> uriVariables,
+                                              final PagingAndSortingDto pagingAndSorting,
+                                              final ParameterizedTypeReference<T> responseType) {
+        return getWebClient(false)
+                .get()
+                .uri(uriBuilder -> uriBuilder.path(path).queryParams(queryParams).build(uriVariables))
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                .headers(withPaging(pagingAndSorting))
+                .attributes(clientRegistrationId("elite2-api"))
```

Comment:
In your WebClientConfig you can do

webClient.setDefaultClientRegistrationId("elite2-api");

which means you don't have to specify the clientRegistrationId every time, but it's not important :)